addons/sale_timesheet/models/sale_order.py

```diff
@@ -19,7 +19,7 @@ class SaleOrder(models.Model):
 
     visible_project = fields.Boolean('Display project', compute='_compute_visible_project', readonly=True)
     project_id = fields.Many2one(
-        'project.project', 'Project', domain=[('billable_type', 'in', ('no', 'task_rate')), ('analytic_account_id', '!=', False)],
+        'project.project', 'Project', domain="[('billable_type', 'in', ('no', 'task_rate')), ('analytic_account_id', '!=', False), ('company_id', '=', company_id)]",
```

Comment:
@RomainLibert ,
in the master sale order and project, both have company_id required so it can not be blank any case so make it sharable does not make any sense. 
sale: https://github.com/odoo/odoo/blob/master/addons/sale/models/sale.py#L176
project: https://github.com/odoo/odoo/blob/master/addons/project/models/project.py#L168
Thanks